src/Parser/Support.idr

```diff
@@ -138,64 +138,65 @@ getEsc "SP" = Just '\SP'
-           ([], rest) => assert_total (escape' rest)
-           (ds, rest) => pure $ cast !(toHex 1 (reverse ds)) ::
-                                 !(assert_total (escape' rest))
+escape' : List Char -> List Char -> Maybe (List Char)
+escape' _ [] = pure []
+escape' escapeChars (x::xs)
+    = assert_total $ if escapeChars `isPrefixOf` (x::xs)
+         then case drop (length escapeChars) (x::xs) of
+                   ('\\' :: xs) => pure $ '\\' :: !(escape' escapeChars xs)
+                   ('&' :: xs) => pure !(escape' escapeChars xs)
+                   ('a' :: xs) => pure $ '\a' :: !(escape' escapeChars xs)
+                   ('b' :: xs) => pure $ '\b' :: !(escape' escapeChars xs)
+                   ('f' :: xs) => pure $ '\f' :: !(escape' escapeChars xs)
+                   ('n' :: xs) => pure $ '\n' :: !(escape' escapeChars xs)
+                   ('r' :: xs) => pure $ '\r' :: !(escape' escapeChars xs)
+                   ('t' :: xs) => pure $ '\t' :: !(escape' escapeChars xs)
+                   ('v' :: xs) => pure $ '\v' :: !(escape' escapeChars xs)
+                   ('\'' :: xs) => pure $ '\'' :: !(escape' escapeChars xs)
+                   ('"' :: xs) => pure $ '"' :: !(escape' escapeChars xs)
+                   ('x' :: xs) => case span isHexDigit xs of
+                                       ([], rest) => escape' escapeChars rest
+                                       (ds, rest) => pure $ cast !(toHex 1 (reverse ds)) ::
+                                                             !(escape' escapeChars rest)
+                   ('o' :: xs) => case span isOctDigit xs of
+                                       ([], rest) => escape' escapeChars rest
+                                       (ds, rest) => pure $ cast !(toOct 1 (reverse ds)) ::
+                                                             !(escape' escapeChars rest)
+                   xs => case span isDigit xs of
+                              ([], (a :: b :: c :: rest)) =>
+                                case getEsc (fastPack (the (List _) [a, b, c])) of
+                                     Just v => Just (v :: !(escape' escapeChars rest))
+                                     Nothing => case getEsc (fastPack (the (List _) [a, b])) of
+                                                     Just v => Just (v :: !(escape' escapeChars (c :: rest)))
+                                                     Nothing => escape' escapeChars xs
+                              ([], (a :: b :: [])) =>
+                                case getEsc (fastPack (the (List _) [a, b])) of
+                                     Just v => Just (v :: [])
+                                     Nothing => escape' escapeChars xs
+                              ([], rest) => escape' escapeChars rest
+                              (ds, rest) => Just $ cast (cast {to=Int} (fastPack ds)) ::
+                                              !(escape' escapeChars rest)
```

Comment:
This is big enough that it should probably be in its own function

Reply:
It's going to be fully refactored in multi-line PR, therefore we can leave it for now.

Reply:
sounds good